pointgrey_camera_driver/src/nodelet.cpp

```diff
@@ -367,17 +367,62 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
       }
       catch(std::runtime_error& e)
       {
+        // Something terrible has happened, so let's just disconnect and reconnect to see if we can recover.
         NODELET_ERROR("%s", e.what());
-        ///< @todo Look into readding this
-        /*try{
-          // Something terrible has happened, so let's just disconnect and reconnect to see if we can recover.
-          pg_.disconnect();
-          ros::Duration(1.0).sleep(); // sleep for one second each time
-          pg_.connect();
-          pg_.start();
-        } catch(std::runtime_error& e2){
-          NODELET_ERROR("%s", e2.what());
-        }*/
+
+        // Try disconnecting from the camera
```

Comment:
There's some inter-dependency with `connectCb`.

I think I actually need to change the way we recover because https://github.com/ros-drivers/pointgrey_camera_driver/blob/master/pointgrey_camera_driver/src/nodelet.cpp#L315 needs to be checked in every iteration. That might break the lazy publisher logic, that stops grabbing images when the subscriber disconnects from the topic.

I'm going to test also that and make sure that it works.

Regarding the pattern you propose, I guess it's possible, moving the start/stop, connect/disconnect logic from the connectCb to the devicePoll loop, adding initial steps/states.

Sounds good?